### PR TITLE
Pool2D OP

### PR DESCRIPTION
**Pull request by lambda7xx**
**Description of changes:**
- Pool2D OP

**Related Issues:**

Linked Issues:
- Issue #1072

Issues closed by this PR:
- Closes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/flexflow/FlexFlow/1182)

### CODE REVIEW

File: lib/kernels/src/cuda/pool_2d_kernels.cu
Comment (reyna-abhyankar) [inline lib/kernels/src/cuda/pool_2d_kernels.cu:1]: Also do hip kernel?

File: lib/runtime/src/ops/pool_2d.cc
Comment (reyna-abhyankar) [inline lib/runtime/src/ops/pool_2d.cc:1]: Split signature from register task

**Review by reyna-abhyankar — CHANGES_REQUESTED** [review]

File: lib/runtime/src/ops/pool_2d.cc [hunk 1]
```diff
@@ -1,288 +1,48 @@
...
-using PCG::Node;
-bool operator==(Pool2DParams const &lhs, Pool2DParams const &rhs) {
-  return lhs.kernel_h == rhs.kernel_h && lhs.kernel_w == rhs.kernel_w &&
-         lhs.stride_h == rhs.stride_h && lhs.stride_w == rhs.stride_w &&
-         lhs.padding_h == rhs.padding_h && lhs.padding_w == rhs.padding_w &&
-         lhs.pool_type == rhs.pool_type && lhs.activation == rhs.activation;
-}
+static DeviceSpecific<Pool2dPerDeviceState>
+    init_task_impl(TaskArgumentAccessor const &acc) {
+  NOT_IMPLEMENTED();
```
Comment (reyna-abhyankar) [inline lib/runtime/src/ops/pool_2d.cc:28]: @lambda7xx is this still not implemented?

**Comment by lambda7xx** [comment]
*[`lib/runtime/src/ops/pool_2d.cc` line 28 at r2](https://reviewable.io/reviews/flexflow/FlexFlow/1182#gh-1369088886:-Nhd0wjo7P_7Y9T9aUn6:b4wfniu) ([raw file](https://github.com/flexflow/FlexFlow/blob/2928870914f07658e47ec7045b94c566888641ea/lib/runtime/src/ops/pool_2d.cc#L28)):*
<details><summary><i>Previously, reyna-abhyankar (Reyna Abhyankar) wrote…</i></summary><blockquote>

@lambda7xx is this still not implemented?
</blockquote></details>

remove the ` NOT_IMPLEMENTED();`

**Review by reyna-abhyankar — APPROVED** [review]

**Pull request merged**